src/lib/engine/graph.js

```diff
@@ -46,20 +46,18 @@ export function dependencyList(graph, spaceId) {
 // This could be optimized for filtering the graph by the space subset
 export function dependencyTree(oGraph, graphFilters) {
   const {spaceId, metricId, onlyHead, notHead} = graphFilters
-  if (onlyHead) {
-    return [[metricId, 0]]
-  } else {
-    let graph = oGraph
-    if (spaceId) { graph = _space.subset(oGraph, spaceId) }
+  let graph = oGraph
+  if (spaceId) { graph = _space.subset(oGraph, spaceId) }
 
-    let bGraph = basicGraph(graph)
-    if (metricId) { bGraph = bGraph.subsetFrom(metricId) }
+  let bGraph = basicGraph(graph)
+  if (metricId) { bGraph = bGraph.subsetFrom(metricId) }
 
-    const nodes =  bGraph.nodes.map(n => [n.id, n.maxDistanceFromRoot])
-    if (notHead){
-      return nodes.filter(e => (e[0] !== metricId))
-    } else {
-      return nodes
-    }
+  const nodes = bGraph.nodes.map(n => [n.id, n.maxDistanceFromRoot])
+  if (onlyHead) {
```

Comment:
The onlyHead check was on top because there are expensive calculations here that it would be nice not to do when it is just the head.